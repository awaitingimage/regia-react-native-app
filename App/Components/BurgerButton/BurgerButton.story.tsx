import { storiesOf } from "@storybook/react-native";
import * as React from "react";

import BurgerButton from "./BurgerButton";

storiesOf("BurgerButton", module)
  .add("Default", () => (
    <BurgerButton onPress={(e) => {alert("burger button press"); }} />
  ))
  .add("Black", () => (
    <BurgerButton color="black" onPress={(e) => {alert("burger button press"); }} />
  ));
