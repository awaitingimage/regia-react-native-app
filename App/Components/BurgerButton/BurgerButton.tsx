import * as React from "react";
import { GestureResponderEvent, StyleProp, Text, TouchableOpacity, ViewStyle } from "react-native";
import Metrics from "../../Themes/Metrics";
import Icon from "react-native-vector-icons/Ionicons";
import styles from "./Style";

interface Props {
  style?: StyleProp<ViewStyle>;
  onPress?: (event: GestureResponderEvent) => void;
  color?: string;
}

const defaultProps = {
  style: styles.container,
  onPress: () => alert("menu button"),
  color: "black",
};

const BurgerButton: React.SFC<Props> = ({style, onPress, color}: Props) =>  (
  <TouchableOpacity accessibilityLabel={"menu"} onPress={onPress} style={style}>
    <Icon name="ios-menu" style={[styles.icon, {color}]} />
  </TouchableOpacity>
);

BurgerButton.defaultProps = defaultProps;

export default BurgerButton;
