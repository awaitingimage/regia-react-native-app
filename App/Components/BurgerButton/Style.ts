import { StyleSheet } from "react-native";
import { Colors, Fonts, Metrics } from "../../Themes";

export default StyleSheet.create({
  container: {
    position: "absolute", 
    right: Metrics.baseMargin, 
    top: Metrics.baseMargin
  },
  icon: {
    padding: Metrics.baseMargin,
    fontSize: 32,
  },
});
