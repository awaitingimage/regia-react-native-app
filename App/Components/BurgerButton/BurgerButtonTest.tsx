/// <reference types="@types/jest" />
import * as React from "react";
import * as renderer from "react-test-renderer";
import BurgerButton from "./BurgerButton";

test("BurgerButton component renders correctly", () => {
  const tree = renderer.create(<BurgerButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
