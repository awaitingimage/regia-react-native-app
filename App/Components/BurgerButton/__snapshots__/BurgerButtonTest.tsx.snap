// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel="menu"
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "position": "absolute",
      "right": 10,
      "top": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "fontSize": 32,
            "padding": 10,
          },
          Object {
            "color": "black",
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï‡ƒ
  </Text>
</View>
`;
